linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1286,55 +1490,24 @@ private[emitter] object CoreJSLib {
             val ctor = {
               val arg = varRef("arg")
               val i = varRef("i")
-              MethodDef(static = false, Ident("constructor"),
-                  paramList(arg), {
-                Block(
-                    {
-                      if (useClassesForRegularClasses) Apply(Super(), Nil)
-                      else Skip()
-                    },
-                    If(typeof(arg) === str("number"), {
-                      // arg is the length of the array
-                      Block(
-                          privateFieldSet("u", New(ArrayRef, arg :: Nil)),
-                          For(let(i, 0), i < arg, i.++, {
-                            BracketSelect(This() DOT "u", i) := componentZero
-                          })
-                      )
-                    }, {
-                      // arg is a native array that we wrap
-                      privateFieldSet("u", arg)
-                    })
-                )
-              })
-            }
-
-            val getAndSet = if (arrayIndexOutOfBounds != CheckedBehavior.Unchecked) {
-              val i = varRef("i")
-              val v = varRef("v")
-
-              val boundsCheck = {
-                If((i < 0) || (i >= (This() DOT "u" DOT "length")),
-                    genCallHelper("throwArrayIndexOutOfBoundsException", i),
-                    Skip())
-              }
-
-              List(
-                  MethodDef(static = false, Ident("get"), paramList(i), {
+              MethodDef(static = false, Ident("constructor"), paramList(arg), {
+                if (useClassesForRegularClasses) {
+                  Apply(Super(), arg :: Nil)
+                } else {
+                  If(typeof(arg) === str("number"), {
+                    // arg is the length of the array
                     Block(
-                        boundsCheck,
-                        Return(BracketSelect(This() DOT "u", i))
-                    )
-                  }),
-                  MethodDef(static = false, Ident("set"), paramList(i, v), {
-                    Block(
-                        boundsCheck,
-                        BracketSelect(This() DOT "u", i) := v
+                        privateFieldSet("u", New(ArrayRef, arg :: Nil)),
+                        For(let(i, 0), i < arg, i.++, {
+                          BracketSelect(This() DOT "u", i) := Null()
+                        })
                     )
+                  }, {
+                    // arg is a native array that we wrap
+                    privateFieldSet("u", arg)
```

Comment:
OK. Fair enough :) Then maybe we should put it in a helper (taking the tree for the component zero) where we can also add a comment explaining this? (Unless of course this helper will immediately become useless in the next commit). The same comment applies to `clone` I think.

Somewhat unrelated: Have you considered using [`Array.prototype.fill`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill) for this in ES6?

Reply:
Good point. I have extracted in a helper.


> Somewhat unrelated: Have you considered using [`Array.prototype.fill`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill) for this in ES6?

I have noticed it and considered using it. I've quickly verified that it works, but I haven't performed any benchmarks with it. I'd rather avoid implementing it now, at the same time as all the other changes, because I would want to perform independent benchmarks for that change on its own.